elyra/pipeline/handlers.py

```diff
@@ -97,37 +100,38 @@ async def post(self, *args, **kwargs):
 
 
 class PipelineComponentHandler(HttpErrorMixin, APIHandler):
-    """Handler to expose method calls to retrieve pipelines editor component configuration"""
+    """Handler to expose method calls to retrieve pipelines editor component_id configuration"""
 
     @web.authenticated
     async def get(self, processor):
-        self.log.info(f'Retrieving pipeline components for {processor}')
+        self.log.info(f'Retrieving pipeline components for: {processor} runtime')
+
         if PipelineProcessorManager.instance().is_supported_runtime(processor) is False:
             raise web.HTTPError(400, f"Invalid processor name '{processor}'")
 
-        components = await PipelineProcessorManager.instance().get_components(processor)
-        json_msg = json.dumps(components)
+        components: List[Component] = await PipelineProcessorManager.instance().get_components(processor)
 
         self.set_status(200)
         self.set_header("Content-Type", 'application/json')
-        self.finish(json_msg)
+        self.finish(ComponentRegistry.to_canvas_palette(components))
 
 
 class PipelineComponentPropertiesHandler(HttpErrorMixin, APIHandler):
-    """Handler to expose method calls to retrieve pipeline component properties"""
-
-    component_registry: ComponentRegistry = ComponentRegistry()
+    """Handler to expose method calls to retrieve pipeline component_id properties"""
 
     @web.authenticated
     async def get(self, processor, component_id):
-        self.log.info(f'Retrieving pipeline component properties for component {component_id}')
+        self.log.info(f'Retrieving pipeline component_id properties for component_id: {component_id}')
+
         if PipelineProcessorManager.instance().is_supported_runtime(processor) is False:
             raise web.HTTPError(400, f"Invalid processor name '{processor}'")
 
-        properties = ComponentRegistry().get_properties(processor, component_id)
-        json_msg = json.dumps(properties)
+        if not component_id:
+            raise web.HTTPError(400, f"Invalid component_id id '{component_id}'")
 
-        self.set_status(200)
+        component: List[Component] = \
+            await PipelineProcessorManager.instance().get_component_properties(processor, component_id)
```

Comment:
This implies a list of components is returned.  Yet, if I'm tracking things properly, a `Component` is returned.  Which I also find confusing (noted previously) since this is all about getting a component's properties.